### ISSUE TITLE
Consolidate all user preferences/configuration values into one file, or at least remove them from client_state.xml

### ISSUE THREAD

**Issue by Ageless93** [BOINC/boinc#2714]
While helping someone figure out why something doesn't work as it should on his BOINC, I again found that I have to steer this person to various different files in the data directory, to check on values. At this moment BOINC and BOINC Manager store the user preferences/configuration files in four different places:
- client_state.xml
- global_prefs.xml
- global_prefs_override.xml
- cc_config.xml

I'm wondering if it isn't better to have all of the configuration/preferences values in one file, or in the least remove them from client_state.xml and into cc_config.xml or another (new?) configuration file. 

Because now when the user has to manually make a change in client_state.xml and he makes a mistake, all work he's got stored in it will be removed and although we have a backup file (client_state_prev.xml), this isn't used immediately as a backup. Instead, as I found out, it's just overwritten with the new client_state.xml contents without all the lost work. This happens whether you're quick to exit & restart the client or not. 

I also understand that this requires a substantial rewrite of the client and Manager, and may require that the PMC agree on it. And that it's therefore not for a next version.  

But just asking how much it would take?

**Comment by AenBleidd** [comment]
In case of Windows (I'm not sure about linux and MacOS) Manager stores its settings in Windows Registry so to the list above you can add one more point

**Comment by AenBleidd** [comment]
BTW, next ticket #321 is also connected to this

**Comment by Ageless93** [comment]
> client_state.xml is not a config file. 

I agree, but with a lot of the entries at the bottom in there, it has become one:
```
<platform_name>windows_x86_64</platform_name>
<core_client_major_version>7</core_client_major_version>
<core_client_minor_version>13</core_client_minor_version>
<core_client_release>0</core_client_release>
<user_run_request>2</user_run_request>
<user_run_prev_request>1</user_run_prev_request>
<user_gpu_request>2</user_gpu_request>
<user_gpu_prev_request>3</user_gpu_prev_request>
<user_network_request>2</user_network_request>
<new_version_check_time>1537086553.173491</new_version_check_time>
<all_projects_list_check_time>1536656155.955572</all_projects_list_check_time>
<language>en_GB</language>
<client_version_check_url>https://boinc.berkeley.edu/download.php?xml=1</client_version_check_url>
<alt_platform>windows_intelx86</alt_platform>
<host_venue>work</host_venue>
```

**Comment by RichardHaselgrove** [comment]
I used this feature extensively when testing my recent tuning of the preferences dialog. But I noticed that loading the preferences dialog when the override file was disabled led to a long delay - long enough to trigger the 'Communicating with client - please wait' popup - suggesting to my mind that the active web preference set is being refetched anyway. But the global_prefs file on disk isn't updated.

**Comment by Aurum420** [comment]
1. I've never found a reason to edit my client_state.xml file but plenty of reasons to look up values and names in it. Having said that,
2. LHC ATLAS has a problem with how they use LHC_Preferences/Max # WUs and Max # CPUs. computezrmle came up with this solution to give the client proper control:
```
<app_version>
    <app_name>ATLAS</app_name>
    <plan_class>native_mt</plan_class>
    <avg_ncpus>1</avg_ncpus>
    <cmdline>--nthreads 1</cmdline>
</app_version>
```
Followed by a BOINC restart when it overwrites these values into ones client_state.xml file. 
If it's decided to alter these files please assure clients can emulate that level of control.
